rustle_cli/src/main.rs

```diff
@@ -1,3 +1,281 @@
+                                compile_directory(rd, output.clone())
+                            } else {
+                                let e = dir_creation_result.err().unwrap();
+
+                                if let ErrorKind::AlreadyExists = e.kind() {
+                                    compile_directory(rd, output.clone());
+                                } else {
+                                    println!("{}: {}", style("[ERROR]").red(), e);
+                                }
+                            }
+
+                        },
+                        Err(e) => {
+                            println!("{}: {}", style("[ERROR]").red(), e);
+                        }
+                    };
+                } else if md.is_file() {
+                    if let Some(out_file) = cli.output.as_deref() {
+                        let mut location = current_dir().unwrap();
+                        let mut input = location.clone();
+                        let filePath = Path::new(file);
+                        input.push(filePath);
+
+
+                        location.push(out_file);
+
+                        let output = compile_file_to_js(input.as_path(), location.as_path());
+
+                        match output {
+                            Err(e) => {
+                                println!("{}: {}", style("[ERROR]").red(), e);
+                            },
+                            _ => {
+
+                            }
+                        }
+                    } else {
+                        let mut location = current_dir().unwrap();
+                        let mut input = location.clone();
+                        let filePath = Path::new(file);
+                        input.push(filePath);
+
+
+                        location.push(filePath.file_stem().unwrap().to_str().unwrap().to_owned() + ".js");
+
+                        let output = compile_file_to_js(input.as_path(), location.as_path());
+
+                        match output {
+                            Err(e) => {
+                                println!("{}: {}", style("[ERROR]").red(), e);
+                            },
+                            _ => {
+
+                            }
+                        }
+                    }
+                }
+            },
+            Err(e) => {
+                match e.kind() {
+                    ErrorKind::NotFound => {
+                        println!("{}: File or Directory not found", style("[ERROR]").red());
+                    },
+
+                    ErrorKind::PermissionDenied => {
+                        println!("{}: Permission Denied", style("[ERROR]").red());
+                    },
+
+                    _ => {
+                        println!("{}: unexpected error", style("[ERROR]").red());
+                    }
+                }
+            }
+        };
+
+    }
+
+
+    // Parse subcommand
+    match &cli.command {
```

Comment:
I'm not sure why this is duplicated? I'll check again tomorrow

Reply:
I duplicated this to separate the logic for parsing and compiling I also wanted parse to be a subcommand (rustle parse [FILE]) instead of an argument. This could be changed if you wanted.

Reply:
I think for now a simple CLI would work best, without a subcommand, so you can just go `rustle index.rustle`. Maybe for now just comment out this section and we'll look at it in the future?

Reply:
So something like `rustle index.svelte --parse` for parsing?

Reply:
Aight gonna get on with adding that.

Reply:
Pushed a new commit with the changes you requested